sky/backends/cloud_vm_ray_backend.py

```diff
@@ -1362,16 +1362,31 @@ def _retry_zones(
         # Optimization - check if user has non-zero quota for
         # the instance type in the target region. If not, fail early
         # instead of trying to provision and failing later.
-        try:
-            need_provision = to_provision.cloud.check_quota_available(
-                to_provision.region, to_provision.instance_type,
-                to_provision.use_spot)
 
-        except Exception as e:  # pylint: disable=broad-except
-            need_provision = True
-            logger.info(f'Error occurred when trying to check quota. '
-                        f'Proceeding assuming quotas are available. Error: '
-                        f'{common_utils.format_exception(e, use_bracket=True)}')
+        need_provision = True
+        if to_provision.cloud.is_same_cloud(clouds.AWS()):
+            try:
+                need_provision = to_provision.cloud.check_quota_available(
+                    to_provision.region, to_provision.instance_type,
+                    None, to_provision.use_spot)
+
+            except Exception as e:  # pylint: disable=broad-except
+                logger.info(f'Error occurred when trying to check quota. '
+                            f'Proceeding assuming quotas are available. Error: '
+                            f'{common_utils.format_exception(e, use_bracket=True)}')
+        
+        if to_provision.cloud.is_same_cloud(clouds.GCP()) and to_provision.accelerators:
+            try:
+                need_provision = to_provision.cloud.check_quota_available(
+                    to_provision.region, None, list(to_provision.accelerators.keys())[0],
+                    to_provision.use_spot
+                )
+                
+            except Exception as e:
+                logger.info(f'Error occurred when trying to check quota. '
+                            f'Proceeding assuming quotas are available. Error: '
+                            f'{common_utils.format_exception(e, use_bracket=True)}')
```

Comment:
Instead of having multiple if conditions, can we maintain the previous structure and change the method signature to accept just a `Resources` object as a parameter? That way, the high level method signature is clean, and the underlying implementation can use whatever fields it needs from the resources.

Here's a suggested refactoring:

This code block becomes:
```
        try:
            need_provision = to_provision.cloud.check_quota_available(to_provision)

        except Exception as e:  # pylint: disable=broad-except
            need_provision = True
            logger.info(f'Error occurred when trying to check quota. '
                        f'Proceeding assuming quotas are available. Error: '
                        f'{common_utils.format_exception(e, use_bracket=True)}')
```

check_quota_available signature becomes:
```
    def check_quota_available(cls, resources: resources.Resources) -> bool:
```

And then the implementation can extract fields from as required:
e.g., AWS:
```
    def check_quota_available(cls, resources: resources.Resources) -> bool:
        region = resources.region
        instance_type = resources.instance_type
        ...
```
e.g., GCP:
```
    def check_quota_available(cls, resources: resources.Resources) -> bool:
        region = resources.region
        accelerators = resources.accelerators
        ...
```

What do you think?